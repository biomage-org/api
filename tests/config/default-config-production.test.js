// @ts-nocheck
const _ = require('lodash');

const AWS = require('aws-sdk');
const isPromise = require('../../src/utils/isPromise');
const { ACCOUNT_ID } = require('../../src/api.v2/constants');

jest.mock('aws-sdk');
jest.mock('../../src/utils/getLogger');

describe('default-config', () => {
  const OLD_ENV = _.cloneDeep(process.env);

  beforeEach(() => {
    jest.clearAllMocks();
    process.env = OLD_ENV;
  });

  afterAll(() => {
    jest.clearAllMocks();
    process.env = OLD_ENV;
  });

  it('Returns correct values for BIOMAGE production', () => {
    const prodEnvironment = 'production';
    process.env.NODE_ENV = prodEnvironment;
    process.env.K8S_ENV = prodEnvironment;
    process.env.CLUSTER_ENV = prodEnvironment;
    process.env.AWS_DEFAULT_REGION = 'eu-west-1';
    process.env.AWS_ACCOUNT_ID = ACCOUNT_ID.BIOMAGE;

    const userPoolId = 'mockUserPoolId';
    const accountId = 'mockAccountId';

    AWS.CognitoIdentityServiceProvider = jest.fn(() => ({
      listUserPools: {
        promise: jest.fn(() => Promise.resolve(
          { UserPools: [{ id: userPoolId, Name: `biomage-user-pool-case-insensitive-${prodEnvironment}` }] },
        )),
      },
    }));

    AWS.STS = jest.fn(() => ({
      getCallerIdentity: {
        promise: jest.fn(() => Promise.resolve({ Account: accountId })),
      },
    }));

    const defaultConfig = jest.requireActual('../../src/config/default-config');

    const defaultConfigEntries = Object.entries(defaultConfig);
    const filteredEntries = defaultConfigEntries.filter(([, value]) => !isPromise(value));
    const defaultConfigFiltered = Object.fromEntries(filteredEntries);

    expect(defaultConfigFiltered).toMatchSnapshot();
  });

  it('Returns correct values for HMS production', () => {
    jest.unmock('../../src/config/default-config');
    jest.resetModules();

    const prodEnvironment = 'production';
    process.env.NODE_ENV = prodEnvironment;
    process.env.K8S_ENV = prodEnvironment;
    process.env.CLUSTER_ENV = prodEnvironment;
    process.env.AWS_DEFAULT_REGION = 'eu-west-1';
    process.env.AWS_ACCOUNT_ID = ACCOUNT_ID.HMS;

    const userPoolId = 'mockUserPoolId';
    const accountId = 'mockAccountId';

    AWS.CognitoIdentityServiceProvider = jest.fn(() => ({
      listUserPools: {
        promise: jest.fn(() => Promise.resolve(
          { UserPools: [{ id: userPoolId, Name: `biomage-user-pool-case-insensitive-${prodEnvironment}` }] },
        )),
      },
    }));

    AWS.STS = jest.fn(() => ({
      getCallerIdentity: {
        promise: jest.fn(() => Promise.resolve({ Account: accountId })),
      },
    }));

    const defaultConfig = jest.requireActual('../../src/config/default-config');

    const defaultConfigEntries = Object.entries(defaultConfig);
    const filteredEntries = defaultConfigEntries.filter(([, value]) => !isPromise(value));
    const defaultConfigFiltered = Object.fromEntries(filteredEntries);

    expect(defaultConfigFiltered).toMatchSnapshot();
  });
});
