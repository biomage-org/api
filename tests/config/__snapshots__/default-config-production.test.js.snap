// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default-config Returns correct values for BIOMAGE production 1`] = `
Object {
  "adminSub": "032abd44-0cd3-4d58-af21-850ca0b95ac7",
  "api": Object {
    "prefix": "/",
  },
  "awsAccountId": "242905224710",
  "awsBatchIgnoreSSLCertificate": false,
  "awsRegion": "eu-west-1",
  "cachingEnabled": true,
  "clusterEnv": "production",
  "cognitoISP": Object {
    "listUserPools": Object {
      "promise": [MockFunction],
    },
  },
  "corsOriginUrl": Array [
    "https://sandbox.elabjournal.com",
    "https://elabjournal.com",
    "https://localhost:5000",
  ],
  "datadogApiKey": "",
  "datadogAppKey": "",
  "domainName": "localhost:5000",
  "emailDomainName": "https://localhost:5000",
  "pipelineInstanceConfigUrl": "https://raw.githubusercontent.com/hms-dbmi-cellenics/releases/master/production/pipeline.yaml",
  "pipelineNamespace": "pipeline-default",
  "podName": "local",
  "port": 3000,
  "publicApiUrl": "https://api.localhost:5000",
  "rdsSandboxId": "default",
  "sandboxId": "default",
  "workerNamespace": "worker-default",
}
`;

exports[`default-config Returns correct values for HMS production 1`] = `
Object {
  "adminSub": "a01e8bcc-c9a2-4c56-bd66-39de93764be8",
  "api": Object {
    "prefix": "/",
  },
  "awsAccountId": "160782110667",
  "awsBatchIgnoreSSLCertificate": false,
  "awsRegion": "eu-west-1",
  "cachingEnabled": true,
  "clusterEnv": "production",
  "cognitoISP": mockConstructor {
    "CALL_EVENTS_BUBBLE": [MockFunction],
    "MONITOR_EVENTS_BUBBLE": [MockFunction],
  },
  "corsOriginUrl": Array [
    "https://sandbox.elabjournal.com",
    "https://elabjournal.com",
    "https://localhost:5000",
  ],
  "datadogApiKey": "",
  "datadogAppKey": "",
  "domainName": "localhost:5000",
  "emailDomainName": "https://localhost:5000",
  "pipelineInstanceConfigUrl": "https://raw.githubusercontent.com/hms-dbmi-cellenics/releases/master/production/pipeline.yaml",
  "pipelineNamespace": "pipeline-default",
  "podName": "local",
  "port": 3000,
  "publicApiUrl": "https://api.localhost:5000",
  "rdsSandboxId": "default",
  "sandboxId": "default",
  "workerNamespace": "worker-default",
}
`;
